src/components/geometry.js

```diff
@@ -35,6 +41,39 @@ module.exports.Component = registerComponent('geometry', {
 
     // Create new geometry.
     this.geometry = mesh.geometry = system.getOrCreateGeometry(data);
+    if (diffData.mergeTo) {
+      if (!data.mergeTo) { return; }
+      this.mergeTo(data.mergeTo);
+    }
+  },
+
+  mergeTo: function (toEl) {
+    var el = this.el;
+    var mesh = this.el.getObject3D('mesh');
+    var toMesh = toEl.getObject3D('mesh');
+    if (!toEl) {
+      warn('There is not a valid entity to merge the geometry to');
+      return;
+    }
+    if (toEl === el) { return; }
+    if (!toMesh) {
+      warn('Merge is not posible. The entity ' + toEl.id + ' doesn\'t have a mesh');
+      return;
+    }
+
+    if (toMesh.geometry instanceof THREE.Geometry === false) {
+      warn('Geometry merge is only available for THREE.Geometry types');
+      return;
+    }
+
+    if (this.data.skipCache === false) {
+      warn('Cached geometries are not allowed to merge. Set skipCache to true');
+      return;
+    }
+
+    mesh.parent.updateMatrixWorld();
+    toMesh.geometry.merge(mesh.geometry, mesh.matrixWorld);
```

Comment:
Might be nice to emit an event on the target entity (e.g, `geometry-merged`). I might want to switch the final geometry to a buffer geometry after everything is merged, would be nice to have a declarative way of doing so